packages/firebase_remote_config/ios/Classes/FirebaseRemoteConfigPlugin.m

```diff
@@ -97,12 +104,28 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                     }
                   }];
   } else if ([@"RemoteConfig#activate" isEqualToString:call.method]) {
-    BOOL newConfig = [[FIRRemoteConfig remoteConfig] activateFetched];
-    NSDictionary *parameters = [self getConfigParameters];
-    NSMutableDictionary *resultDict = [[NSMutableDictionary alloc] init];
-    resultDict[@"newConfig"] = [NSNumber numberWithBool:newConfig];
-    resultDict[@"parameters"] = parameters;
-    result(resultDict);
+    [[FIRRemoteConfig remoteConfig]
+        activateWithCompletion:^(BOOL changed, NSError *_Nullable error) {
+          BOOL newConfig = @true;
+          if (error) {
+            if (error.userInfo && error.userInfo[@"ActivationFailureReason"] != nil &&
+                [error.userInfo[@"ActivationFailureReason"] containsString:@"already activated"]) {
+              newConfig = @false;
+            } else {
+              FlutterError *flutterError;
+              flutterError = [FlutterError errorWithCode:@"activateFailed"
+                                                 message:error.userInfo[@"ActivationFailureReason"]
+                                                 details:nil];
+              result(flutterError);
+            }
+          }
```

Comment:
I think there ought to be an `else` block wrapped around the rest of the `activateWithCompletion ` code block (i.e line 123-127).

Reply:
I think that's not exactly correct - there are two error cases, one which returns a regular result but noting it's "already activated" and one which is a real error type that falls into calling code error handling blocks

So the if (error) block is differentiating between those two, then the following block executes no matter what (assuming an error wasn't tossed out from the if(error && reallyWasAnError) case

Similar to the above note though, if that's not immediately clear, it's a defect. I'll think if there is a way to shape it up differently or at minimum add a quick comment explaining

Reply:
ah, I see. The reason why I raise that is the callback will be called twice if the error is not "ActivationFailureReason" if I'm not mistaken? This may result in another error/possible duplicate response in the dart code.  A `return;` after `result(flutterError);` I think ought to do the trick I think 🤔

Reply:
...and *that* is a real problem. Would have missed that even as I'm polishing it. I'll see if I can throw some test cover in there as well on error conditions to expose it since it's obvious in hindsight but was in my blind spot. Thanks

Reply:
I added explanatory comments, and fixed the "missing return in the error case" issue you found.
I explored adding cover but there were enough inconsistencies in how lastFetchTime and activation status were coming back that I feel it's best left as a minimal port per @Salakar direction, pending landing of #4186 which will otherwise be a massive collision with this